color/src/tagged.rs

```diff
@@ -178,6 +178,21 @@ impl ColorSpaceTag {
             }
         }
     }
+
+    /// Clip the color's components to fit within the natural gamut of the color space.
+    ///
+    /// See [`ColorSpace::clip`] for more details.
+    pub fn clip(self, src: [f32; 3]) -> [f32; 3] {
+        match self {
+            Self::Srgb => Srgb::clip(src),
+            Self::LinearSrgb => LinearSrgb::clip(src),
+            Self::Oklab => Oklab::clip(src),
+            Self::Oklch => Oklch::clip(src),
+            Self::DisplayP3 => DisplayP3::clip(src),
+            Self::XyzD65 => XyzD65::clip(src),
+            _ => todo!(),
```

Comment:
This isn't ideal, but I think this is *fine*. The expectation is that all three `todo!`s in this file would be fixed as a block

Reply:
Some tags are not yet implemented. To move the concerns (and code duplication) to a single place, perhaps a private `match`-like macro that takes by-name a fallback `ColorSpace` method makes sense for the ergonomics. `ColorSpace` is not currently object-safe so we can't go through a `&dyn ColorSpace`.

Reply:
(Alternatively, we deal with this when more color spaces are implemented and accept the code duplication.)

Reply:
We will probably end up with a macro, but keep in mind some methods do really want custom logic; `convert` is the big one.

Reply:
I suspect for the amount of things, a macro is probably not necessary. But we can discuss that when the last implementations land.